src/vm/vm_core.rs

```diff
@@ -2489,58 +2408,28 @@ mod tests {
     /// FP 1:2
     /// PC 0:0
     fn test_step_for_preset_memory() {
-        let mut vm = VirtualMachine::new(
-            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 134217728]),
-            Vec::new(),
-            true,
-        );
+        let mut vm = vm!(true);
         vm.accessed_addresses = Some(Vec::new());
-        for _ in 0..4 {
-            vm.memory.data.push(Vec::new());
-        }
+
         vm.run_context.pc = MaybeRelocatable::from((0, 0));
         vm.run_context.ap = MaybeRelocatable::from((1, 2));
         vm.run_context.fp = MaybeRelocatable::from((1, 2));
```

Comment:
We could add a macro for RunContext initialization, something like run_context!(0,2,2) that would generate a RunContext from the register's offsets (as their indexes are constant), it will be really helpful for tests on cairo_runner too

Reply:
It would also let us abstract the structure of the run context in tests so we can easily swap it

Reply:
Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/378/commits/1f23cd481892418f40eb3dd4f12a4ae6b2939cb9